helios-client/src/main/java/com/spotify/helios/client/HeliosClient.java

```diff
@@ -655,9 +661,18 @@ public HeliosClient build() {
     }
 
     private static ListeningScheduledExecutorService defaultExecutorService() {
-      final ScheduledExecutorService executor = MoreExecutors.getExitingScheduledExecutorService(
-          (ScheduledThreadPoolExecutor) newScheduledThreadPool(4), 0, SECONDS);
-      return MoreExecutors.listeningDecorator(executor);
+      final int clientCount = clientCounter.incrementAndGet();
+
+      final ThreadFactory threadFactory = new ThreadFactoryBuilder()
+          .setNameFormat("helios-client-" + clientCount + "-thread-%d")
```

Comment:
I'm not 100% sure, but it seems like `Executors.defaultThreadFactory()` will do this type of naming. Perhaps just create one instance for all `HeliosClient`s instead of creating a new instance of a `ThreadFactory` for every `HeliosClient` instance? Or will that not be enough in some way?